samples/net/cloud/mqtt_azure/prj.conf

```diff
@@ -33,6 +33,7 @@ CONFIG_MBEDTLS_ENABLE_HEAP=y
 CONFIG_MBEDTLS_HEAP_SIZE=100000
 CONFIG_MBEDTLS_SSL_MAX_CONTENT_LEN=10240
 CONFIG_MBEDTLS_PEM_CERTIFICATE_FORMAT=y
+CONFIG_MBEDTLS_MAC_SHA1_ENABLED=y
```

Comment:
It looks that Microsoft certificate uses SHA384. It also seems that the server requires ECDHE_RSA key exchange, otherwise it rejects the connection. Can you try with the following configs added?
```
CONFIG_MBEDTLS_MAC_SHA384_ENABLED=y
CONFIG_MBEDTLS_KEY_EXCHANGE_ECDHE_RSA_ENABLED=y
CONFIG_MBEDTLS_ECP_DP_SECP256R1_ENABLED=y
CONFIG_MBEDTLS_ECDH_C=y
CONFIG_MBEDTLS_ECP_C=y
```
I was able to establish TLS session with those configs enabled.

Reply:
Actually, as https://github.com/zephyrproject-rtos/zephyr/pull/73267 was just merged (as I mentioned, a lot of reworks in mbed TLS area recently) it should be now:
```
CONFIG_MBEDTLS_SHA1=y
CONFIG_MBEDTLS_SHA384=y
```
Other configs as stated above.